### PR TITLE
Add e2e CI execution

### PR DESCRIPTION
**Pull request by Kuvesz**
## Description

This PR aims to add very basic CI functionality for e2e tests with creating a kind cluster and running the tests on that. The kind cluster creation is in a separate composite action to make sure it can be used elsewhere and parameters can be set.
An example run can be seen here: https://github.com/banzaicloud/koperator/actions/runs/5592373354/jobs/10224749825

## Type of Change
- [x] Other (CI, e2e testing)

## Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] I have verified this change is not present in other open pull requests
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Review by bartam1 — APPROVED** [review]
Im not a github action expert but it is LGTM!

**Review by panyuenlau — APPROVED** [review]
Same here - not a GHA expert.

---

I found that the logs that we have [in the "run tests" step](https://github.com/banzaicloud/koperator/actions/runs/5598488874/jobs/10238192109?pr=1016#step:5:1) are not quite readable (too many logs), I am wondering if there is a way to only show the detailed logs on failures (for debugging purposes), and only show the high level steps if the tests don't break: e.g.:
```
  > Enter [It] Acquiring K8s config and context @ 07/19/23 11:27:57.652
  < Exit [It] Acquiring K8s config and context @ 07/19/23 11:27:57.655 (4ms)
  > Enter [It] Installing cert-manager Helm chart @ 07/19/23 11:27:57.657
  < Exit [It] Installing cert-manager Helm chart @ 07/19/23 11:28:21.935 (24.278s)
  > Enter [It] Installing zookeeper-operator Helm chart @ 07/19/23 11:28:21.936
  < Exit [It] Installing zookeeper-operator Helm chart @ 07/19/23 11:29:05.016 (43.08s)
  > Enter [It] Installing prometheus-operator Helm chart @ 07/19/23 11:29:05.016
  < Exit [It] Installing prometheus-operator Helm chart @ 07/19/23 11:29:36.455 (31.439s)
  > Enter [It] Installing Koperator Helm chart @ 07/19/23 11:29:36.455
...
```

I believe having so many logs in stdout would slow down the tests execution quite a bit, and in fact, GHA can't show the full logs since it is too much:
```
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```

**Comment by Kuvesz** [comment]
> Same here - not a GHA expert.
> 
> I found that the logs that we have [in the "run tests" step](https://github.com/banzaicloud/koperator/actions/runs/5598488874/jobs/10238192109?pr=1016#step:5:1) are not quite readable, I am wondering if there is a way to only show the detailed logs on failures (for debugging purposes), and only show the high level steps if the tests don't break: e.g.:
> 
> ```
>   > Enter [It] Acquiring K8s config and context @ 07/19/23 11:27:57.652
>   < Exit [It] Acquiring K8s config and context @ 07/19/23 11:27:57.655 (4ms)
>   > Enter [It] Installing cert-manager Helm chart @ 07/19/23 11:27:57.657
>   < Exit [It] Installing cert-manager Helm chart @ 07/19/23 11:28:21.935 (24.278s)
>   > Enter [It] Installing zookeeper-operator Helm chart @ 07/19/23 11:28:21.936
>   < Exit [It] Installing zookeeper-operator Helm chart @ 07/19/23 11:29:05.016 (43.08s)
>   > Enter [It] Installing prometheus-operator Helm chart @ 07/19/23 11:29:05.016
>   < Exit [It] Installing prometheus-operator Helm chart @ 07/19/23 11:29:36.455 (31.439s)
>   > Enter [It] Installing Koperator Helm chart @ 07/19/23 11:29:36.455
> ...
> ```
> 
> I believe having so many logs in stdout would slow down the tests execution quite a bit, and in fact, GHA can't show the full logs since it is too much:
> 
> ```
> This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
> ```

I believe this is exactly the same log we get when running from the terminal, so to show less we'd have to change the logging of the tests themselves? However I'll look into this a bit later on if we can somehow make the logs shorter specifically in the CI, like with a regex or something?

**Comment by panyuenlau** [comment]
> so to show less we'd have to change the logging of the tests themselves

Yea I think we can remove these two lines: https://github.com/banzaicloud/koperator/blob/master/Makefile#L107-L108.

---

Do you think these two lines are strictly needed, @pregnor? Asking since you are the author of the make command so there might be something that I don't see for

**Comment by bartam1** [comment]
I think most of the log messages are helm installation related. 
We should investigate what should we remove first if we want shortening the log

**Comment by Kuvesz** [comment]
> I think most of the log messages are helm installation related. We should investigate what should we remove first if we want shortening the log

Looked into it a bit and it seems to me that `show-node-events` or `trace` isn't really a culprit here. A reference to what `show-node-events` does: https://github.com/onsi/ginkgo/blob/a0d57ac22cad3d0678054d5f16ad86261e1f21f7/types/config.go#L331 (Trace is on the previous parameter in the same line, so I won't put another link here.)

So I'm thinking maybe removing the verbose option might be the solution if we want shorter logs? On failures we're gonna have a trace anyway and if everything is fine we might not need all the logs we can get.

**Comment by pregnor** [comment]
Yeah in general we need to resolve 2 things.
Logs too long and being truncated (we need a less verbose mode for passing, only be verbose on error).
The line formatting needs to be updated, the prefix is too long.
Tickets pending.

File: tests/e2e/koperator_suite_test.go [hunk 1]
```diff
@@ -54,12 +54,12 @@ var _ = BeforeSuite(func() {
...
-	//testInstall()
+	testInstall()
 	testInstallZookeeperCluster()
 	testInstallKafkaCluster("../../config/samples/simplekafkacluster.yaml")
 	testUninstallKafkaCluster()
 	testInstallKafkaCluster("../../config/samples/simplekafkacluster_ssl.yaml")
 	testUninstallKafkaCluster()
 	testUninstallZookeeperCluster()
-	//testUninstall()
+	testUninstall()
```
Comment (pregnor) [inline tests/e2e/koperator_suite_test.go:64]: **Question**: Unrelated to the PR, but why are the producer/consumer tests not enabled here BTW?

**Review by pregnor — APPROVED** [review]

File: tests/e2e/koperator_suite_test.go (hunk 1 above)
Comment (Kuvesz) [reply tests/e2e/koperator_suite_test.go:64]: No idea, probably whoever merged the last PR (Might have been me) didn't work on those and overwrote this file.

Comment (pregnor) [reply tests/e2e/koperator_suite_test.go:64]: I see it will be fixed in #1017.

**Comment by pregnor** [comment]
> > so to show less we'd have to change the logging of the tests themselves
> 
> Yea I think we can remove these two lines: https://github.com/banzaicloud/koperator/blob/master/Makefile#L107-L108.
> 
> ---
> 
> Do you think these two lines are strictly needed, @pregnor? Asking since you are the author of the make command so there might be something that I don't see

Sorry I haven't addressed this.
I would be fine, they were included on Márton's request and I rather wanted to fail in this direction than the other one.

**Pull request merged**